ui/app/adapters/job.js

```diff
@@ -159,4 +166,98 @@ export default class JobAdapter extends WatchableNamespaceIDs {
       },
     });
   }
+
+  runAction(job, action, allocID) {
+    let messageBuffer = '';
+
+    const protocol = window.location.protocol === 'https:' ? 'wss:' : 'ws:';
+    const region = this.system.activeRegion;
+    const applicationAdapter = getOwner(this).lookup('adapter:application');
+    const prefix = `${
+      applicationAdapter.host || window.location.host
+    }/${applicationAdapter.urlPrefix()}`;
+
+    const wsUrl =
+      `${protocol}//${prefix}/job/${job.get('id')}/action` +
+      `?namespace=*&action=${action.name}&allocID=${allocID}&task=${action.task.name}&group=${action.task.taskGroup.name}&tty=true&ws_handshake=true` +
```

Comment:
I'm not sure if `namespace=*` will work in this endpoint, and even if it does I think we want to more precise here and use the job's namespace to avoid any unexpected requests.